### PR TITLE
Hotfix filter_reference_dates

### PR DESCRIPTION
**Pull request by adrian-lison**
## Description

This PR solves a bug that occurred when using `remove_days` in `filter_reference_dates` on data with missing reference dates, but that would only throw an error if using both `include_days` and `remove_days`.

### CODE REVIEW

File: R/preprocess.R [hunk 1]
```diff
@@ -404,8 +404,7 @@ enw_filter_report_dates <- function(obs, latest_date, remove_days) {
 #'
 #' @param include_days if \code{earilest_date} is not given, the number
 #' of reference dates to include, ending with the latest reference
-#' date included once report dates have been removed. If specified
-#' this is indexed to `latest_date` or `remove_days`.
+#' date included (determined by `latest_date` or `remove_days`).
```
Comment (adrian-lison) [inline R/preprocess.R:407]: I am not 100% sure what "once report dates have been removed" meant, but I think it was wrong. The report dates smaller than the earliest date were only removed in the end.

**Review by seabbs — COMMENTED** [review]
Nice catch - looks good. Hopefully retriggered CI will all pass now.

**Review by seabbs — APPROVED** [review]
Nice catch - looks good. Hopefully retriggered CI will all pass now.

**Comment by seabbs** [comment]
/document

I think the CI is just failing because of #360 and so this can be merged. Will keep an eye on the CI over the next few days and hot fix any problems

**Review by seabbs — APPROVED** [review]
LGTM. Ty ❤️

**Pull request merged**